lib/ansible/modules/package_facts.py

```diff
@@ -376,6 +376,31 @@ def get_package_details(self, package):
         return dict(zip(self.atoms, package.split()))
 
 
+class APK(CLIMgr):
+
+    CLI = 'apk'
+    atoms = ['name', 'version']
+
+    def list_installed(self):
+        rc, out, err = module.run_command([self._cli, 'info', '-v'])
+        if rc != 0 or err:
+            raise Exception("Unable to list packages rc=%s : %s" % (rc, err))
+        return out.splitlines()
+
+    def get_package_details(self, package):
+        raw_pkg_details = {}
+        for line in package.splitlines():
+            m = re.match(r"([\w ].*?)-([0-9-\.r]+)", line.decode("utf-8"))
+            if m:
+                raw_pkg_details['Name'] = m.group(1)
+                raw_pkg_details['Version'] = m.group(2)
```

Comment:
Why store these values in a dict, with the right key names, but the wrong case, and then later, build a new dict with the right case?

Probably best to just do it lowercase here, and then just return it.

Reply:
I used PACMAN function as starter and this is how it is handled.
reviewed in latest